tmt/hardware.py

```diff
@@ -672,8 +672,13 @@ def from_specification(
             name: str,
             raw_value: str,
             original_constraint: Optional['Constraint[Any]'] = None,
-            allowed_operators: Optional[list[Operator]] = None
+            allowed_operators: Optional[list[Operator]] = None,
+            unit: Optional[Any] = 'bytes'
             ) -> T:
+        if not isinstance(
+                UNITS(raw_value),
+                pint.Quantity):  # type: ignore[reportUnnecessaryIsInstance,unused-ignore]
+            raw_value = str(pint.Quantity(raw_value, unit))
```

Comment:
I would recommend doing it *after* `_from_specification()` call, inspecting the returned constraint. Doing it *before* has the disadvantage of various input types, for example, your test will not trigger for `memory: "1024"`, because `raw_value` would be a string. On the other hand, *after* the call, the `value` attribute of constraint returned by `_from_specification()` would always be a `Quantity` instance, and much safer to work with, with no exceptions.